ruff_cli/src/printer.rs

```diff
@@ -148,6 +148,27 @@ impl<'a> Printer<'a> {
             }
             return Ok(());
         }
+        // Prints a count of the number of violations for each rule if statistics are
+        // enabled.
+        if matches!(self.log_level, LogLevel::Statistics) {
+            let mut stdout = BufWriter::new(io::stdout().lock());
+            let mut violations = diagnostics
+                .messages
+                .iter()
+                .map(|message| message.kind.rule())
+                .collect::<Vec<_>>();
+            violations.sort();
+            violations.dedup();
+            for rule in violations {
+                let count = diagnostics
+                    .messages
+                    .iter()
+                    .filter(|message| message.kind.rule() == rule)
+                    .count();
+                writeln!(stdout, "Rule {} has {count} violation(s).", rule.code())?;
```

Comment:
I think by default, for now, we should try to adhere to Flake8's format -- number of violations, then code, then default message, with consistent columnar width for the first column:

```
...
2     W391 blank line at end of file
8     W605 invalid escape sequence '\.'
2     W606 'async' and 'await' are reserved keywords starting with Python 3.7
2     YTT101 `sys.version[:3]` referenced (python3.10), use `sys.version_info`
2     YTT102 `sys.version[2]` referenced (python3.10), use `sys.version_info`
5     YTT103 `sys.version` compared to string (python3.10), use `sys.version_info`
4     YTT201 `sys.version_info[0] == 3` referenced (python4), use `>=`
2     YTT202 `six.PY3` referenced (python4), use `not six.PY2`
2     YTT203 `sys.version_info[1]` compared to integer (python4), compare `sys.version_info` to tuple
2     YTT204 `sys.version_info.minor` compared to integer (python4), compare `sys.version_info` to tuple
2     YTT301 `sys.version[0]` referenced (python10), use `sys.version_info`
5     YTT302 `sys.version` compared to string (python10), use `sys.version_info`
2     YTT303 `sys.version[:1]` referenced (python10), use `sys.version_info`
```